plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/rule/RewriteVectorDistanceFunction.java

```diff
@@ -0,0 +1,161 @@
+    {
+        Optional<ParameterizedExpression> leftExpression = rewrite(captures.get(LEFT_ARGUMENT), context);
+        if (leftExpression.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Optional<ParameterizedExpression> rightExpression = rewrite(captures.get(RIGHT_ARGUMENT), context);
+        if (rightExpression.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(new JdbcExpression(
+                "%s %s %s".formatted(leftExpression.get().expression(), operator, rightExpression.get().expression()),
+                ImmutableList.<QueryParameter>builder()
+                        .addAll(leftExpression.get().parameters())
+                        .addAll(rightExpression.get().parameters())
+                        .build(),
+                new JdbcTypeHandle(
+                        Types.DOUBLE,
+                        Optional.of("double"),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty())));
+    }
+
+    private static Optional<ParameterizedExpression> rewrite(ConnectorExpression expression, RewriteContext<ParameterizedExpression> context)
+    {
+        if (expression instanceof Constant constant) {
+            Type elementType = ((ArrayType) constant.getType()).getElementType();
+            Block value = (Block) constant.getValue();
+            StringJoiner vector = new StringJoiner(",", "'[", "]'");
+            for (int i = 0; i < value.getPositionCount(); i++) {
+                if (value.isNull(i)) {
+                    return Optional.empty();
+                }
+                double doubleValue = elementType.getDouble(value, i);
+                if (!isSupportedVector(doubleValue)) {
+                    return Optional.empty();
+                }
+                vector.add(Double.toString(doubleValue));
+            }
+            return Optional.of(new ParameterizedExpression(vector.toString(), ImmutableList.of()));
+        }
+        if (expression instanceof Call call && call.getFunctionName().equals(CAST_FUNCTION_NAME)) {
+            ConnectorExpression argument = getOnlyElement(call.getArguments());
+            if (argument instanceof Variable variable) {
+                JdbcTypeHandle typeHandle = ((JdbcColumnHandle) context.getAssignment(variable.getName())).getJdbcTypeHandle();
+                // TODO type.equals("vector") should be improved to support pushdown on vector type which is installed in other schemas
+                if (!typeHandle.jdbcTypeName().map(type -> type.equals("vector")).orElse(false)) {
+                    return Optional.empty();
+                }
+                return Optional.of(new ParameterizedExpression(quoted(variable.getName()), ImmutableList.of()));
```

Comment:
nit: Instead of using `quoted` here ? Can we use `context.rewriteExpression()` which would add the corresponding quotes ?